jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/FrameFlusher.java

```diff
@@ -147,10 +155,9 @@ protected Action process() throws Throwable
             if (closedCause != null)
                 throw closedCause;
 
-            // Succeed entries from previous call to process
-            // and clear batchBuffer if we wrote it.
-            if (succeedEntries() && batchBuffer != null)
-                BufferUtil.clear(batchBuffer);
```

Comment:
I think clearing the batchBuffer here is incorrect, it produces test failures because it is only partially sending the content. I have removed it in the changes

this line of code just happened to never be called because we succeed entries in the callback first, so the succeed entries in the if statement here always returns false

Reply:
OK... we just have to make sure that a clear or reset is done on the buffer at some stage so we don't end up with an empty batchBuffer with pos==limit==capacity, so we can't add anything to it.   Correct use of BufferUtil normally protects against this... but just check to make sure it does.